packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -325,12 +395,41 @@ export class EngineReportingAgent<TContext = any> {
       });
     }
 
+    if (this.options.endpointUrl) {
+      this.logger.warn(
+        '[deprecated] The `endpointUrl` option within `engine` has been renamed to `tracesEndpointUrl`.',
+      );
+    }
+    this.tracesEndpointUrl =
+      (this.options.endpointUrl ||
+        this.options.tracesEndpointUrl ||
```

Comment:
I'd suggest preferring the newer thing over the back-compat thing if for some reason both are provided? If you passed in the new one, that means you did so more recently...